reflection.md

```diff
@@ -15,4 +18,40 @@ Write a few brief paragraphs that discuss:
    you implement it _or_ what features of Scala would prevent you from
    implementing it? (You don't have to write code for this part. You could say
    "I would use literal extension to..." or "Scala's rules for valid
-   identifiers prevent...")
\ No newline at end of file
+   identifiers prevent...")
+
+
+The easiest things to say are things with repetition. For me, regular expressions can be strong when using union operators with the Kleene star. One can then specify patterns that can be arbitrarily long (with the Kleene star) and relatively complex using the union operator within the Kleene star. In fact, the family of operators `<*>`, `<+>`, and `{n}` are all related to doing things with repetition. As a result, we can be very specific about our repetitions.
+
+On the other hand, it seems that it is fairly difficult to write regular expressions when there are patterns that aren't easily represented using repetition. For example, the code we were given had the following example:
+
+```
+val digit = '0' || '1' || '2' || '3' || '4' || '5' || '6' || '7' || '8' || '9' 
+```
+
+This code made me a bit sad because there is an obvious pattern, but we were forced to union 10 literals. In general, union and concatenation operations can create clunky regular expressions. Not that they by themselves are bothersome, but when things get cumbersome, I believe these two operations are deeply involved.
+
+Another trait about our DSL that slightly bothers me is how we have to explicitly use parentheses to correctly express our regular expression:
+
+```
+val pattern = "42" || ( ('a' <*>) ~ ('b' <+>) ~ ('c'{3}))
+```
+
+In this code, we are forced to put three pairs of parentheses to write this regular expression. It would be interesting to see if we can create some sort of rules that allows us to remove parentheses. Some parentheses can be good for clarifying the regular expressions, but for those that seem redundant, removing parentheses could be nice.
+
+Besides removing parentheses, there are three other things I would consider modifying. One is to further simplify the syntax of the DSL. For example, it would be nice to write just a `*` in front of a regular expression like:
+
+```
+("42" || "6")*
+```
+to say I want 0 or more copies of what is inside the parentheses. However, I would like to mention that this change is probably not possible in Scala. It would probably be fairly difficult or impossible) to get Scala to recognize the `*` operator in the code above.
```

Comment:
Hmmm I also thought of this and rejected the idea because it could be difficult to teach Scala to recognize * in this case as a character that represents repetition instead of multiplication. However, I think that this could actually work! Maybe we could teach Scala that if there is no object / argument that comes after the * sign, that the * should be understood as a means to repeat what came before it. Thus, if no object / argument comes after the * sign, the * sign should not be interpreted as multiplication (which makes sense, because what would be multiplied together if there was an argument before * but no argument after * ?).